upgrade/steps-helpers.go

```diff
@@ -351,3 +352,113 @@ func getRepoExpectedLocation(ctx Context, cwd, repoPath string) (string, error)
 
 	return filepath.Join(ctx.GoPath, "src", expectedLocation), nil
 }
+
+// Fetch the expected upgrade target from github. Return a list of open upgrade issues,
+// sorted by semantic version. The list may be empty.
+//
+// The second argument represents a message to describe the result. It may be empty.
+func GetExpectedTarget(ctx Context, name, upstreamOrg string) (*UpstreamUpgradeTarget, string, error) {
+	if ctx.TargetVersion != nil {
+		return &UpstreamUpgradeTarget{Version: ctx.TargetVersion}, "", nil
+	}
+	if !ctx.InferVersion {
+		return getExpectedTargetLatest(ctx, name, upstreamOrg)
+	}
+	return getExpectedTargetFromIssues(ctx, name)
+}
+
+func getExpectedTargetLatest(ctx Context, name, upstreamOrg string) (*UpstreamUpgradeTarget, string, error) {
+	latest := exec.CommandContext(ctx, "gh", "release", "list",
+		"--repo="+upstreamOrg+"/"+ctx.UpstreamProviderName,
+		"-L=1",
+		"--exclude-drafts",
+		"--exclude-pre-releases")
+	bytes := new(bytes.Buffer)
+	latest.Stdout = bytes
+	err := latest.Run()
+	if err != nil {
+		return nil, "", err
+	}
+	release := []struct {
+		Title string `json:"TITLE"`
+		Type  string `json:"TYPE"`
+	}{}
+	err = json.Unmarshal(bytes.Bytes(), &release)
+	if err != nil {
+		return nil, "", err
+	}
+
+	contract.Assert(len(release) == 1)
+	contract.Assert(release[0].Type == "Latest")
+	v, err := semver.NewVersion(release[0].Title)
+	if err != nil {
+		return nil, "", err
+	}
+	return &UpstreamUpgradeTarget{Version: v}, "", nil
+}
+
+func getExpectedTargetFromIssues(ctx Context, name string) (*UpstreamUpgradeTarget, string, error) {
```

Comment:
Is this the same as what `getExpectedTarget` used to be?

Reply:
Yes, now `GetExpectedTarget` calls either `getExpectedTargetLatest` or `getExpectedTargetFromIssues` depending on the flags set